packages/graph-schema-utils/src/formatters/json/types.ts

```diff
@@ -14,28 +20,113 @@ export type GraphSchemaJsonStruct = {
   relationshipTypes: RelationshipTypeJsonStruct[];
   nodeObjectTypes: NodeObjectTypeJsonStruct[];
   relationshipObjectTypes: RelationshipObjectTypeJsonStruct[];
+  constraints: ConstraintJsonStruct[];
+  indexes: IndexJsonStruct[];
 };
 
-export type NodeLabelJsonStruct = { $id: string; token: string };
+export type NodeLabelJsonStruct = {
+  $id: string;
+  token: string;
+  properties: PropertyJsonStruct[];
+};
 
-export type RelationshipTypeJsonStruct = { $id: string; token: string };
+export type RelationshipTypeJsonStruct = {
+  $id: string;
+  token: string;
+  properties: PropertyJsonStruct[];
+};
 
 export type NodeObjectTypeJsonStruct = {
   $id: string;
   labels: { $ref: string }[];
-  properties: PropertyJsonStruct[];
 };
 
 export type RelationshipObjectTypeJsonStruct = {
   $id: string;
   type: { $ref: string };
   from: { $ref: string };
   to: { $ref: string };
-  properties: PropertyJsonStruct[];
+};
+
+export type ConstraintJsonStruct = {
+  $id: string;
+  name: string;
+  constraintType: ConstraintType;
+  entityType: EntityType;
+  nodeLabel?: { $ref: string };
+  relationshipType?: { $ref: string };
+  properties: { $ref: string }[];
+};
+
+export type NodeLabelConstraintJsonStruct = ConstraintJsonStruct & {
+  nodeLabel: { $ref: string };
+};
```

Comment:
Should this say:

```suggestion
export type NodeLabelConstraintJsonStruct = ConstraintJsonStruct & {
  entityType: "node";
  nodeLabel: { $ref: string };
};
```

? Or, should we get rid of `entityType` on `CosntraintJsonStruct` as we can infer this from the presence of `nodeLabel` or `relationshipType`?

Reply:
Also should we be more strict that no relationship is specified? I saw you did something similar with lookup indexes
```suggestion
export type NodeLabelConstraintJsonStruct = ConstraintJsonStruct & {
  entityType: "node";
  nodeLabel: { $ref: string };
  relationshipType: undefined;
};
```